quadriga/src/main/java/edu/asu/diging/quadriga/web/service/impl/GraphCreationServiceImpl.java

```diff
@@ -0,0 +1,240 @@
+                objectNodeId = createNodesAndEdges((RelationEvent) relation.getObject(), graphNodes, graphEdges,
+                        uniqueNodes, eventGraphId);
+            } else {
+                objectNodeId = createSubjectOrObjectNode(graphNodes, (AppellationEvent) relation.getObject(),
+                        uniqueNodes, GraphNodeType.OBJECT, eventGraphId);
+            }
+        } else {
+            logger.error("An object is missing in one of the relations for EventGraph: " + eventGraphId);
+        }
+
+        if(subjectNodeId != null && predicateNodeId != null) {
+            createEdge(graphEdges, subjectNodeId, predicateNodeId, eventGraphId);
+        }
+        
+        if(objectNodeId != null && predicateNodeId != null) {
+            createEdge(graphEdges, objectNodeId, predicateNodeId, eventGraphId);
+        }
+
+        return predicateNodeId;
+    }
+
+    @Override
+    public String createPredicateNode(List<GraphData> graphNodes, AppellationEvent event, String eventGraphId) {
+        GraphNodeData predicateNode = createNode(event, GraphNodeType.PREDICATE, eventGraphId);
+        graphNodes.add(predicateNode);
+        return predicateNode.getId();
+    }
+
+    @Override
+    public String createSubjectOrObjectNode(List<GraphData> graphNodes, AppellationEvent event,
+            Map<String, GraphNodeData> uniqueNodes, GraphNodeType graphNodeType, String eventGraphId) {
+        String sourceUri = event.getTerm().getInterpretation().getSourceURI();
+        GraphNodeData node;
+        
+        if (sourceUri != null && uniqueNodes.containsKey(sourceUri)) {
+            node = uniqueNodes.get(sourceUri);
+            node.getEventGraphIds().add(eventGraphId);
+            return node.getId();
+        }
+
+        node = createNode(event, graphNodeType, eventGraphId);
+        graphNodes.add(node);
+        uniqueNodes.put(sourceUri, node);
+        
+        if(sourceUri != null && !sourceUri.equals("") && node.getAlternativeUris() != null) {
+            node.getAlternativeUris()
+            .stream()
+            .filter(nullableAltUri -> nullableAltUri != null)
+            .filter(alternativeUri -> !alternativeUri.equals("") && !alternativeUri.equals(sourceUri))
+                .forEach(alternativeUriValue -> uniqueNodes.put(alternativeUriValue, node));
+        }
+        
+        return node.getId();
+    }
+
+    @Override
+    public GraphNodeData createNode(AppellationEvent event, GraphNodeType graphNodeType, String eventGraphId) {
+        
+        GraphNodeData node = new GraphNodeData();
+        node.setId(new ObjectId().toString());
+        node.setGroup(graphNodeType.getGroupId());
+        node.setEventGraphIds(new ArrayList<String>(Collections.singletonList(eventGraphId)));
+
+        String sourceURI = event.getTerm().getInterpretation().getSourceURI();
+
+        if (sourceURI != null && sourceURI.contains("www.digitalhps.org")) {
```

Comment:
strings should not be hard-coded but configuration values (in most cases)

Reply:
still hard-coded

Reply:
do u want me to add it through config.properties?

Reply:
yes

Reply:
?